mysqld_exporter.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"database/sql"
+	_ "github.com/go-sql-driver/mysql"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "mysql"
+)
+
+var (
+	listenAddress = flag.String("web.listen-address", ":9104", "Address to listen on for web interface and telemetry.")
+	metricPath    = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
+)
+
+type Exporter struct {
+	dsn                        string
+	mutex                      sync.RWMutex
+	duration                   prometheus.Gauge
+	totalScrapes, errorScrapes prometheus.Counter
+	metrics                    map[string]prometheus.Gauge
+}
+
+// return new empty exporter
+func NewMySQLExporter(dsn string) *Exporter {
+	return &Exporter{
+		dsn: dsn,
+		duration: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      "exporter_last_scrape_duration_nanoseconds",
+			Help:      "The last scrape  duration.",
+		}),
+		totalScrapes: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "exporter_scrapes_total",
+			Help:      "Current total mysqld scrapes.",
+		}),
+		errorScrapes: prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
That code doesn't look right, you want to make errorScrape a Gauge and use .Set() to set it to 0 or 1